connect/runtime/src/test/java/org/apache/kafka/connect/integration/ExampleConnectIntegrationTest.java

```diff
@@ -159,4 +207,41 @@ private boolean checkForPartitionAssignment() {
             return false;
         }
     }
+
+    /**
+     * Confirm that the requested number of workers is up and running.
+     *
+     * @param numWorkers the number of online workers
+     * @return true if at least {@code numWorkers} are up; false otherwise
+     */
+    private Optional<Boolean> assertWorkersUp(int numWorkers) {
```

Comment:
There are cases where you might want to have `orElse(true)`. Indeed no such cases are written here, but strictly speaking the fact that you had a failure doesn't mean neither `true` nor `false`. That's what's represented by the empty optional and I feel conceptually it's correct.

Reply:
One use case is if you want to negate the result of your assertion method. In this case, `orElse` should be `true` because no answer does not mean correct answer.